### DISCUSSION TITLE
How to know API description?

### DISCUSSION THREAD

**Discussion by shimoshida** [gavanderhoorn/comet_rpc#11] [Q&A] [answered]
Thank you for developing this very useful tool!

I would like to develop using this tool, but I am thinking of extending it as it does not currently support register writing. 
However, I do not have the information necessary for implementation, such as RpcId information, and I do not know how to find out what to do.
E.g., `RcpId.REGVALRD( = "247")` exists, but `REGVALSET` does not exist...

Could you please share how you obtained this information when you developed this tool?

**Comment by gavanderhoorn** [comment, marked as answer]
> Could you please share how you obtained this information when you developed this tool?

please see #4 for a previous discussion about this specific aspect.

> E.g., `RcpId.REGVALRD( = "247")` exists, but `REGVALSET` does not exist...

Not when I looked at this, no. But you also don't necessarily need it.

(numeric) registers are treated as system variables by some parts of the Fanuc controller OS, so you can use `VMIP_WRITEVA` instead to write to registers.

See how [Dominh does this](https://github.com/gavanderhoorn/dominh/blob/385116e04ec10955c0f5af47f747befe89811c08/src/dominh/registers.py) for some examples.

**Reply by shimoshida** [reply]
Thank you very much for your very detailed explanation! I understand now.

**Reply by gavanderhoorn** [reply]
If you do end up finding a proper `REGVALSET` (who knows, perhaps Fanuc added one): I'd be interested in Pull Requests extending the current implementation.
